main.go

```diff
@@ -50,8 +50,7 @@ func main() {
 	if *outname == "-" {
 		outfile = os.Stdout
 	} else {
-		flag := os.O_WRONLY | os.O_TRUNC
-		file, err := os.OpenFile(*outname, flag, 0666)
+		file, err := os.Create(*outname)
```

Comment:
This will use `os.O_RDWR` instead of `os.O_WRONLY`.

But I see the deficiency you're trying to address--the existing code won't create the file if it doesn't already exist.  In my use, I think I mainly used the default file redirection.

Just add `os.O_CREATE` to `flag` and this will be good-to-go.